docs/testing.rst

```diff
@@ -0,0 +1,57 @@
+Testing
+=======
+
+Privly's Safari Extension supports both unit testing and integration
+testing. The tests can be run either locally or on SauceLabs.
+
+Unit Testing
+------------
+The extension's unit testing is supported with the help of `Jasmine
+tests <http://jasmine.github.io/2.3/introduction.html>`_ run by the
+`Karma Test Runner <http://karma-runner.github.io/0.13/index.html>`_.
+
+To run the unit tests, we need to first install the packages. To do
+this, navigate to the :code:`privly.safariextension/privly-applications/test`
+directory and run the command :code:`npm install -g karma-cli && npm install`.
+Finally navigate to the :code:`test` directory to run the tests.
+
+To run the unit tests locally, run the command
+:code:`./run_each.sh ../privly.safariextension/privly-applications/test/karma.conf.js`
+
+To run the unit tests on SauceLabs, run the command
+:code:`./run_each.sh ../privly.safariextension/privly-applications/test/karma.conf-ci.js`
+
+Integration Testing
+-------------------
+The extension presently does not support integration testing with
+the help of `Selenium <http://www.seleniumhq.org>`_. So, to make
+sure that the extension is functioning as expected, we need to
+manually follow the steps given below to ensure the same.
+
+To test injected link functionality:
+
+- Navigate to :code:`http://test.privly.org/test_pages/url_parameters.html` on the browser.
```

Comment:
Add a few URLs you will visit to ensure they still inject properly, like the "host page tests" on test.privly.org.